### PR TITLE
Fix package set step by upgrading npm in CI

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

As explained here: https://github.com/purescript/purescript/pull/4299#issuecomment-1112666399

Also updates Node to 14 since that will be EOF in a few months.

---

**Checklist:**

- [ ] Added a file to CHANGELOG.d for this PR (see CHANGELOG.d/README.md)
- [ ] Added myself to CONTRIBUTORS.md (if this is my first contribution)
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
CI fails with:
```
Run npm i -g npm@8.8.0
  npm i -g npm@8.8.0
  "../ci/fix-home stack --haddock exec ../ci/build-package-set.sh"
  shell: bash --noprofile --norc -e -o pipefail {0}
  env:
    CI_RELEASE: false
    STACK_VERSION: 2.7.3
/__t/node/14.19.1/x64/bin/npm -> /__t/node/14.19.1/x64/lib/node_modules/npm/bin/npm-cli.js
/__t/node/14.19.1/x64/bin/npx -> /__t/node/14.19.1/x64/lib/node_modules/npm/bin/npx-cli.js
+ npm@8.8.0
added 63 packages from 18 contributors, removed 299 packages and updated 138 packages in 7.832s
/__w/_temp/533430fc-67e6-4d4d-906c-b80ebd93975f.sh: line 2: ../ci/fix-home stack --haddock exec ../ci/build-package-set.sh: No such file or directory
Error: Process completed with exit code 127.
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -89,7 +89,12 @@ jobs:
         # into which stack places all build artifacts. Since we use --haddock
         # in our CI builds, in order to actually get stack to find the purs
         # binary it created, we need to use the flag here as well.
-        run: "../ci/fix-home stack --haddock exec ../ci/build-package-set.sh"
+        #
+        # Moreover, npm has a hook issue that will cause spago to fail to install
+        # We upgrade npm to fix this
+        run: |
+          npm i -g npm@8.8.0
+          "../ci/fix-home stack --haddock exec ../ci/build-package-set.sh"
```
Comment (rhendric) [inline .github/workflows/ci.yml:97]: Lose these quotes.

Comment (sd-yip) [inline .github/workflows/ci.yml:97]:
```suggestion
          ../ci/fix-home stack --haddock exec ../ci/build-package-set.sh
```

**Comment by JordanMartinez** [comment]
Just curious, why were those quotes added in the first place? They also appear here: https://github.com/purescript/purescript/pull/4302/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR80

**Comment by JordanMartinez** [comment]
I'll be out for the next two hours or so, so I won't be able to respond to any updates.

**Comment by rhendric** [comment]
> Just curious, why were those quotes added in the first place? They also appear here: https://github.com/purescript/purescript/pull/4302/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR80

In YAML, a pipe character starts a block quote. For single-line strings, this isn't necessary, and so I defaulted to standard quotation marks in those cases.

This is complicated by the fact that in very many cases, quotation marks are optional in YAML. If you work long enough with YAML, you will encounter one of the cases in which omitting quotation marks gets you into trouble (the [Norway problem](https://hitchdev.com/strictyaml/why/implicit-typing-removed/) is a memorable one), so out of an abundance of caution for those who might cargo-cult after me, I quoted all of the string scalars I put in the workflow files, with either quote or pipe characters. I don't know if other contributors have been so strict.

**Comment by JordanMartinez** [comment]
> > Just curious, why were those quotes added in the first place? They also appear here: https://github.com/purescript/purescript/pull/4302/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR80
> 
> In YAML, a pipe character starts a block quote. For single-line strings, this isn't necessary, and so I defaulted to standard quotation marks in those cases.
> 
> This is complicated by the fact that in very many cases, quotation marks are optional in YAML. If you work long enough with YAML, you will encounter one of the cases in which omitting quotation marks gets you into trouble (the [Norway problem](https://hitchdev.com/strictyaml/why/implicit-typing-removed/) is a memorable one), so out of an abundance of caution for those who might cargo-cult after me, I quoted all of the string scalars I put in the workflow files, with either quote or pipe characters. I don't know if other contributors have been so strict.

Ah, gotcha. I was wondering why that was there.

**Comment by JordanMartinez** [comment]
Once CI builds, this should be ready for review.

**Review by rhendric — APPROVED** [review]

**Pull request merged**